### ISSUE TITLE
[angle] crash on macos

### ISSUE THREAD

**Issue by xiaozhuai** [microsoft/vcpkg#29622]
**Describe the bug**
angle+glfw crash on **macos** after #27444

Here is a demo project [vcpkg_demo](https://github.com/xiaozhuai/vcpkg_demo)

<img width="1091" alt="image" src="https://user-images.githubusercontent.com/4773701/218392323-a9e40af8-a515-44a1-a663-e5cd69c1ec68.png">

Also, theres is a warning when running on macos
```
objc[91749]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x7ffb5026aec8) and /Users/xiaozhuai/work/vcpkg_demo/cmake-build-debug/opengles_angle/vcpkg_demo_opengles_angle (0x10909dbd8). One of the two will be used. Which one is undefined.
```

**Environment**
- OS: MacOS 12.6.3
- Compiler: Apple clang version 14.0.0 (clang-1400.0.29.202)

**Comment by xiaozhuai** [comment]
Hi @jimwang118 , any progress on this? I investigated, but found nothing.

**Comment by jimwang118** [comment]
> Hi @jimwang118 , any progress on this? I investigated, but found nothing.

Can you tell me the specific detailed operation?

**Comment by xiaozhuai** [comment]
@jimwang118 
Check this repo [vcpkg_demo](https://github.com/xiaozhuai/vcpkg_demo)
Comment out other demos. You will need only this `add_subdirectory(opengles_angle)`
And then build and run on macos.

**Comment by jimwang118** [comment]
Ok, I reproduced the problem locally and I'm fixing up this.

**Comment by jimwang118** [comment]
Thanks for posting this issue. This issue has been fixed. So I am closing this issue for now.

**Comment by xiaozhuai** [comment]
@jimwang118 Thanks for the fixing. Could you please give a link to the fixed pr, I can't find it.

**Comment by xiaozhuai** [comment]
@jimwang118 I don't think this issue has been fixed. I've double checkd and there is no pr link to this issue.

**Comment by xiaozhuai** [comment]
Any progress here?

**Comment by xiaozhuai** [comment]
After an in-depth investigation, I finally found the problem. And this will be fixed by #31163
